servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/RetryPolicy.java

```diff
@@ -45,13 +45,8 @@ public void incrementRetryCount(String clientId)
      * Resets the number of retry attempts made by a client. This method is called by the client when retried operation succeeds.
      * @param clientId id of the client that just retried a failed operation and succeeded.
      */
-    public void resetRetryCount(String clientId)
-    {
-        Integer currentRetryCount = this.retryCounts.get(clientId);
-        if (currentRetryCount != null && currentRetryCount.intValue() != 0)
-        {
-            this.retryCounts.put(clientId, 0);
-        }
+    public void resetRetryCount(String clientId) {
```

Comment:
[SpotBugs-P3]
Sequence of calls to java.util.concurrent.ConcurrentHashMap may not be atomic in com.microsoft.azure.servicebus.primitives.RetryPolicy.resetRetryCount(String)